packages/amplify-category-function/src/provider-utils/awscloudformation/utils/environmentVariablesHelper.ts

```diff
@@ -170,55 +171,52 @@ export const ensureEnvironmentVariableValues = async (context: $TSContext) => {
   if (functionNames.length === 0) {
     return;
   }
-  let printedMessage = false;
-
-  for (const functionName of functionNames) {
-    const storedList = getStoredList(functionName);
-    const storedKeyValue = getStoredKeyValue(functionName);
-    for (const item of storedList) {
-      const envName = item.environmentVariableName;
-      const keyName = item.cloudFormationParameterName;
-      if (storedKeyValue.hasOwnProperty(keyName)) continue;
-      if (!printedMessage) {
-        if (yesFlagSet) {
-          const errMessage = `An error occurred pushing an env "${currentEnvName}" due to missing environment variable values`;
-          context.print.error(errMessage);
-          await context.usageData.emitError(new Error(errMessage));
-          exitOnNextTick(1);
-        } else {
-          context.print.info('');
-          context.print.info('Some Lambda function environment variables are defined but missing values.');
-          context.print.info('');
-        }
-        printedMessage = true;
-      }
-
-      const newValueQuestion: inquirer.InputQuestion = {
-        type: 'input',
-        name: 'newValue',
-        message: `Enter the missing environment variable value of ${envName} in ${functionName}:`,
-        validate: input => {
-          if (input.length >= 2048) {
-            return 'The value must be 2048 characters or less';
-          }
-          return true;
-        },
+
+  const funcConfigMissingEnvVars = functionNames
+    .map(funcName => {
+      const storedList = getStoredList(funcName);
+      const keyValues = getStoredKeyValue(funcName);
+      return {
+        funcName,
+        existingKeyValues: keyValues,
+        missingEnvVars: storedList.filter(({ cloudFormationParameterName: keyName }) => !keyValues.hasOwnProperty(keyName)),
       };
-      const { newValue } = await inquirer.prompt(newValueQuestion);
-      setStoredKeyValue(functionName, {
-        ...storedKeyValue,
-        [keyName]: newValue,
+    })
+    .filter(envVars => envVars.missingEnvVars.length);
+
+  if (!funcConfigMissingEnvVars.length) {
+    return;
+  }
+
+  // there are some missing env vars
+
+  if (yesFlagSet) {
```

Comment:
I think we need to fail if we on CI/CD as well. Also can we collapse the var names to a delimited list instead of per env var print out? So we'd see 1 line per function instead of 1 line per var per function.

Reply:
do we have a different signal for determining a CI/CD workflow besides `--yes`?